crml/sylo/src/migration.rs

```diff
@@ -0,0 +1,221 @@
+/* Copyright 2020 Centrality Investments Limited
+*
+* Licensed under the LGPL, Version 3.0 (the "License");
+* you may not use this file except in compliance with the License.
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* You may obtain a copy of the License at the root of this project source code,
+* or at:
+*     https://centrality.ai/licenses/gplv3.txt
+*     https://centrality.ai/licenses/lgplv3.txt
+*/
+
+use crate::{device, groups, inbox, vault};
+use frame_support::{decl_error, decl_module, decl_storage, dispatch::Vec, ensure, weights::SimpleDispatchInfo};
+use frame_system::{ensure_root, ensure_signed};
+use sp_runtime::{DispatchError::BadOrigin, DispatchResult};
+
+pub trait Trait: device::Trait + groups::Trait + inbox::Trait + vault::Trait {}
+
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		MaxDeviceLimitReached,
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as SyloMigration {
+		MigrationAccount: T::AccountId;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin, system = frame_system {
+		type Error = Error<T>;
+
+		#[weight = SimpleDispatchInfo::FixedOperational(0)]
+		pub fn set_migrator_account(origin, account_id: T::AccountId) -> DispatchResult {
+			ensure_root(origin)?;
+			MigrationAccount::<T>::put(account_id);
+			Ok(())
+		}
+
+		#[weight = SimpleDispatchInfo::FixedOperational(0)]
+		pub fn self_destruct(origin) -> DispatchResult {
+			Self::ensure_sylo_migrator(origin)?;
+			MigrationAccount::<T>::kill();
+			Ok(())
+		}
+
+		#[weight = SimpleDispatchInfo::FixedOperational(0)]
+		fn migrate_devices(origin, user_id: T::AccountId, device_ids: Vec<device::DeviceId>) -> DispatchResult {
+			Self::ensure_sylo_migrator(origin)?;
+			ensure!(device_ids.len() <= device::MAX_DEVICES, Error::<T>::MaxDeviceLimitReached);
```

Comment:
this is being checked twice

Reply:
The first check is for the `device_ids` arg and the second is for what's in the storage (which should be 0 but just to be sure)